File: LogFileHandler_src/LogFileHandler/frmLogFileMonitor.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports LogFileManagers

Partial Public Class frmLogFileMonitor
    Inherits Form

    Private TotalFileQty As Integer = 0
    Public Event OnStatusReport(Msg As String)
    Public Event OnInternalError(ex As Exception)
    Public Event OnFileAmount(qty As Integer)
    Public Event OnFileAmountChanged(qty As Integer)
    Private WithEvents _LogFileManagers As LogFileManagers = New LogFileManagers()

    Public Delegate Sub ProcessToSetTotalQtyInvoke(ByVal Qty As Integer)

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub ProcessToParserLogFiles()
        Try
            _LogFileManagers.WriteDataToLocalFiles("Start log file process")
            DisplayMessage("Start log file process")
            _LogFileManagers.StartParserLogFiles()
        Catch ex As Exception
        End Try
    End Sub

    Private Sub FileCompletedHandler(ByVal sMessage As String) Handles _LogFileManagers.OnFileCompleted
        Try
            DisplayMessage(sMessage)
        Catch
        End Try
    End Sub

    Public Delegate Sub DisplayMessageInvoke(ByVal Msg As String)
    Public Delegate Sub DisplayMessageExceptionInvoke(ByVal ex As Exception)

    Private Sub DisplayMessage(ByVal Msg As String)
        Try
            If Me.InvokeRequired Then
                Me.BeginInvoke(New DisplayMessageInvoke(AddressOf DisplayMessage), Msg)
            Else
                ShowMsg(String.Format("{0:yyyy-MM-dd HH:mm:ss.ffff}", DateTime.Now) & " " & Msg & Environment.NewLine)
            End If

        Catch ex As Exception
        End Try
    End Sub

    Private Sub DisplayMessage(ByVal ex As Exception)
        Try
            If Me.InvokeRequired Then
                Me.BeginInvoke(New DisplayMessageExceptionInvoke(AddressOf DisplayMessage), ex)
            Else
                ShowMsg(String.Format("{0:yyyy-MM-dd HH:mm:ss.ffff}", DateTime.Now) & " " & ex.Message & Environment.NewLine & ex.StackTrace & Environment.NewLine)
            End If

        Catch exx As Exception
        End Try
    End Sub

    Private Sub ShowMsg(ByVal Msg As String)
        Try
            Me.richTextBox1.AppendText(Msg)
        Catch exx As Exception
        End Try
    End Sub

    Private Sub ProcessToSetTotalQty(ByVal Qty As Integer) Handles _LogFileManagers.OnFileAmount
        Try
            If Me.InvokeRequired Then
                Me.BeginInvoke(New ProcessToSetTotalQtyInvoke(AddressOf SetTotalQty), Qty)
            Else
                SetTotalQty(Qty)
            End If

        Catch exx As Exception
        End Try
    End Sub

    Private Sub SetTotalQty(ByVal TotalQty As Integer)
        Try
            Me.lblTotalQty.Text = TotalQty.ToString()
            TotalFileQty = TotalQty
        Catch ex As Exception
        End Try
    End Sub

    Public Delegate Sub ProcessToSetRemainingQtyInvoke(ByVal Qty As Integer)

    Private Sub ProcessToSetRemainingQty(ByVal Qty As Integer) Handles _LogFileManagers.OnFileAmountChanged
        Try
            If Me.InvokeRequired Then
                Me.BeginInvoke(New ProcessToSetRemainingQtyInvoke(AddressOf SetRemainingQty), Qty)
            Else
                SetRemainingQty(Qty)
            End If
        Catch ex As Exception
        End Try
    End Sub

    Private Sub SetRemainingQty(ByVal RemainingQty As Integer)
        Try
            TotalFileQty = TotalFileQty - RemainingQty
            Me.lblRemaining.Text = TotalFileQty.ToString()
        Catch exx As Exception
        End Try
    End Sub

    Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
        Try
            Me.Text = Directory.GetCurrentDirectory()
            Me.lblTotalQty.Text = "0"
            Me.lblRemaining.Text = "0"
        Catch ex As Exception
            ShowMsg(ex.Message)
        End Try
    End Sub

    Private Sub btnStart_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnStart.Click
        Try
            ProcessToParserLogFiles()
        Catch ex As Exception
            ShowMsg(ex.Message)
        End Try
    End Sub

End Class

File: LogFileHandler(vb)/LogFileHandler/Forms/frmLogFileMonitor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using static LogFileManagers;

public partial class frmLogFileMonitor : Form
{
    public frmLogFileMonitor()
    {

        InitializeComponent();
    }

    private void ProcessToParserLogFiles()
    {
        try
        {
            LogFileManagers.WriteDataToLocalFiles("Start log file process");
            DisplayMessage("Start log file process");
            LogFileManagers.ParserLogFiles();
        }
        catch (Exception ex)
        {

        }
    }

    private void FileCompletedHandler(string sMessage)
    {
        try
        {
            DisplayMessage(sMessage);
        }
        catch
        {
        }
    }

    public delegate void DisplayMessageInvoke(string Msg);
    public delegate void DisplayMessageExceptionInvoke(Exception ex);
    private void DisplayMessage(string Msg)
    {
        try
        {
            if (this.InvokeRequired)
            {
                this.BeginInvoke(new DisplayMessageInvoke(DisplayMessage), Msg);
            }
            else
            {
                ShowMsg(string.Format("{0:yyyy-MM-dd HH:mm:ss.ffff}", DateTime.Now) + " " + Msg + Environment.NewLine);
            }
        }
        catch (Exception ex)
        {
        }
    }
    private void DisplayMessage(Exception ex)
    {
        try
        {
            if (this.InvokeRequired)
            {
                this.BeginInvoke(new DisplayMessageExceptionInvoke(DisplayMessage), ex);
            }
            else
            {
                ShowMsg(string.Format("{0:yyyy-MM-dd HH:mm:ss.ffff}", DateTime.Now) + " "
                + ex.Message + Environment.NewLine
                + ex.StackTrace + Environment.NewLine);
            }
        }
        catch (Exception exx)
        {
        }
    }

    private void ShowMsg(string Msg)
    {
        try
        {
            this.richTextBox1.AppendText(Msg);
        }
        catch (Exception exx)
        {
        }
    }

    int TotalFileQty = 0;
    public delegate void ProcessToSetTotalQtyInvoke(int Qty);
    private void ProcessToSetTotalQty(int Qty)
    {
        try
        {
            if (this.InvokeRequired)
            {
                this.BeginInvoke(new ProcessToSetTotalQtyInvoke(SetTotalQty), Qty);
            }
            else
            {
                SetTotalQty(Qty);
            }
        }
        catch (Exception exx)
        {
        }
    }
    private void SetTotalQty(int TotalQty)
    {
        try
        {
            this.lblTotalQty.Text = TotalQty.ToString();
            TotalFileQty = TotalQty;
        }
        catch (Exception ex)
        {
        }
    }
    public delegate void ProcessToSetRemainingQtyInvoke(int Qty);
    private void ProcessToSetRemainingQty(int Qty)
    {
        try
        {
            if (this.InvokeRequired)
            {
                this.BeginInvoke(new ProcessToSetRemainingQtyInvoke(SetRemainingQty), Qty);
            }
            else
            {
                SetRemainingQty(Qty);
            }
        }
        catch (Exception ex)
        {
        }
    }
    private void SetRemainingQty(int RemainingQty)
    {
        try
        {
            TotalFileQty = TotalFileQty - RemainingQty;
            this.lblRemaining.Text = TotalFileQty.ToString();
        }
        catch (Exception exx)
        {
        }
    }


    private bool INITLogFileManager()
    {
        bool flag = false;
        try
        {
            LogFileManagers.OnFileCompleted += new FileCompletedHandler(FileCompletedHandler);
            LogFileManagers.OnStatusReport += new StatusReportHandler(DisplayMessage);
            LogFileManagers.OnInternalError += new ErrorHandler(DisplayMessage);
            LogFileManagers.OnFileAmount += new OnFileAmountHandler(ProcessToSetTotalQty);
            LogFileManagers.OnFileAmountChanged += new OnFileAmountChangedHandler(ProcessToSetRemainingQty);

            flag = true;
        }
        catch (Exception ex)
        {
            flag = false;
        }
        return flag;
    }

    private void UninstallEvents()
    {
        LogFileManagers.OnFileCompleted -= new FileCompletedHandler(FileCompletedHandler);
        LogFileManagers.OnStatusReport -= new StatusReportHandler(DisplayMessage);
        LogFileManagers.OnInternalError += new ErrorHandler(DisplayMessage);
        LogFileManagers.OnFileAmount -= new OnFileAmountHandler(ProcessToSetTotalQty);
        LogFileManagers.OnFileAmountChanged -= new OnFileAmountChangedHandler(ProcessToSetRemainingQty);
    }

    private void Form1_Load(object sender, EventArgs e)
    {
        try
        {
            this.Text = Directory.GetCurrentDirectory();
            this.lblTotalQty.Text = "0";
            this.lblRemaining.Text = "0";
            INITLogFileManager();
        }
        catch (Exception ex)
        { ShowMsg(ex.Message); }
    }

    private void btnStart_Click(object sender, EventArgs e)
    {
        try
        {
            ProcessToParserLogFiles();
        }
        catch (Exception ex)
        {
            ShowMsg(ex.Message);
        }
    }
}

